### PR TITLE
treewide: Run through checkpatch.pl and fix errors.

### PR DESCRIPTION
**Pull request by neheb**
Signed-off-by: Rosen Penev <rosenp@gmail.com>

Tested under OpenWrt.

### CODE REVIEW

File: smb2pdu.c [hunk 1]
```diff
@@ -1738,7 +1745,7 @@ static int parse_durable_handle_context(struct cifsd_tcp_conn *conn,
 	int i, err = 0;
 	uint64_t persistent_id = 0;
 	int req_op_level;
-	char *durable_arr[] = {"DH2C", "DHnC", "DH2Q", "DHnQ",
+	const char *const durable_arr[] = {"DH2C", "DHnC", "DH2Q", "DHnQ",
```
Comment (namjaejeon) [inline smb2pdu.c]: Thanks for your patch! There are build error reports from travis-CI. You added const twice.

File: smb1pdu.c [hunk 2]
```diff
@@ -4515,7 +4528,7 @@ static int query_fs_info(struct cifsd_work *work)
 		break;
 	case SMB_QUERY_FS_DEVICE_INFO:
 		/* query fs info device info response is 0 word and 8 bytes */
-		cifsd_debug("GOT SMB_QUERY_FS_DEVICE_INFO\n");
+		cifsd_debug("%s\n", "GOT SMB_QUERY_FS_DEVICE_INFO");
```
Comment (namjaejeon) [inline smb1pdu.c]: Hm.. Could you explain why you changed like this ?

File: smb2pdu.c (hunk 1 above)
Comment (neheb) [reply smb2pdu.c]: Yeah. It was suggested that I do it like that. Without the const after the *, the array itself contains just plain char *.

File: smb1pdu.c (hunk 2 above)
Comment (neheb) [reply smb1pdu.c]: Suggested by checkpatch.pl. I believe GCC's -Wformat=2 also has a similar warning.

File: smb1pdu.c [hunk 3]
```diff
@@ -5433,7 +5446,7 @@ static int set_path_info(struct cifsd_work *work)
 		cifsd_err("info level = %x not implemented yet\n",
 				info_level);
 		pSMB_rsp->hdr.Status.CifsError = NT_STATUS_NOT_IMPLEMENTED;
-		return -ENOSYS;
+		return -ENOTSUP;
```
Comment (namjaejeon) [inline smb1pdu.c]: undeclared error. -ENOTSUP => -EOPNOTSUPP

File: smb1pdu.c [hunk 4]
```diff
@@ -7006,7 +7019,7 @@ static int set_file_info(struct cifsd_work *work)
 		cifsd_err("info level = %x not implemented yet\n",
 				info_level);
 		rsp->hdr.Status.CifsError = NT_STATUS_NOT_IMPLEMENTED;
-		return -ENOSYS;
+		return -ENOTSUP;
```
Comment (namjaejeon) [inline smb1pdu.c]: should change it to -EOPNOTSUPP.

Comment (namjaejeon) [inline smb1pdu.c]: Should change it to -EOPNOTSUPP.

File: smb1pdu.c [hunk 5]
```diff
@@ -7594,7 +7608,7 @@ int smb_query_info(struct cifsd_work *work)
 
 	if (st.mode & S_ISVTX)
 		attr |=  (ATTR_HIDDEN | ATTR_SYSTEM);
-	if (!(st.mode & S_IWUGO))
+	if (!(st.mode & 0222))
```
Comment (namjaejeon) [inline smb1pdu.c]: Could you exlpain why you used constant variable instead of macro ?

File: smb2pdu.c [hunk 6]
```diff
@@ -3339,16 +3344,16 @@ static int buffer_check_err(int reqOutputBufferLength,
...
 		}
+
+		cifsd_debug("Buffer Overflow\n");
+		rsp->hdr.Status = NT_STATUS_BUFFER_OVERFLOW;
+		rsp->hdr.smb2_buf_length = cpu_to_be32(
+					sizeof(struct smb2_hdr) - 4
+					+ reqOutputBufferLength);
+		rsp->OutputBufferLength = cpu_to_le32(
+						reqOutputBufferLength);
+		return 0;
```
Comment (namjaejeon) [inline smb2pdu.c]: maybe, we can remove return 0; line.

File: smb_common.c [hunk 7]
```diff
@@ -534,7 +534,7 @@ enum SHARED_MODE_ERRORS {
 	FILE_DELETE_ERROR,
 };
 
-static const char *shared_mode_errors[] = {
+static const char *const shared_mode_errors[] = {
```
Comment (namjaejeon) [inline smb_common.c]: maybe, Adding *const can cause build error.

File: smb1pdu.c (hunk 5 above)
Comment (neheb) [reply smb1pdu.c]: checkpatch.pl complains about it. Maybe the kernel community feels that the digits are clearer.

File: smb1pdu.c (hunk 4 above)
Comment (neheb) [reply smb1pdu.c]: Done

File: smb1pdu.c (hunk 3 above)
Comment (neheb) [reply smb1pdu.c]: Ah I was missing a P. Changed to the latter.

File: smb_common.c (hunk 7 above)
Comment (neheb) [reply smb_common.c]: Unlikely. Here's from linux-next:

git grep "static\ const\ char\ \*\ const" | wc -l
7122

File: smb2pdu.c (hunk 6 above)
Comment (neheb) [reply smb2pdu.c]: Done

File: smb2pdu.c (hunk 1 above)
Comment (sergey-senozhatsky) [reply smb2pdu.c]: What really would be great is turning this into 'static'.

-ss

File: smb1pdu.c (hunk 5 above)
Comment (sergey-senozhatsky) [reply smb1pdu.c]: These S_IWHVCNSDEFKJHUI  cleanups are really good.

-ss

File: smb1pdu.c (hunk 2 above)
Comment (sergey-senozhatsky) [reply smb1pdu.c]: No. We now have extra vsprintf() to handle %s.

-ss

**Comment by sergey-senozhatsky** [comment]
Okay... So the parts which I would take in as **separate** patches:

- mode cleanups (those S_IWUGO are hard to follow indeed) : yes
- ENOSYS cleanups : yes
- const char const : not sure, maybe.

-ss

File: smb2pdu.c (hunk 1 above)
Comment (sergey-senozhatsky) [reply smb2pdu.c]: static durable_arr[]

<pre>
Updated 1 path from the index
add/remove: 1/0 grow/shrink: 0/2 up/down: 40/-82 (-42)
Function                                     old     new   delta
durable_arr                                    -      40     +40
smb2_open.cold                              2549    2520     -29
smb2_open                                   9088    9035     -53
Total: Before=135229, After=135187, chg -0.03%
</pre>

-ss

**Comment by namjaejeon** [comment]
@neheb I am okay if you remove cifsd_debug(%s, ""); change from this patch.

**Comment by namjaejeon** [comment]
@neheb Could you please send this pull request to cifsd-next branch ?(namjaejeon:cifsd-next)
I used to push them to master after testing and verifying patches.

**Comment by neheb** [comment]
> @neheb I am okay if you remove cifsd_debug(%s, ""); change from this patch.

Done

Also changed branch.

File: smb2pdu.c (hunk 1 above)
Comment (neheb) [reply smb2pdu.c]: Done

**Comment by namjaejeon** [comment]
@neheb Applied!
Thanks for your work!

https://github.com/cifsd-team/cifsd
https://github.com/cifsd-team/cifsd-tools

 I have generated a new organization called cifsd-team. I decided to
 move cifsd, cifsd-tools repositories here from now on to improve
 the efficiency of collaboration. You can send a pull reqest here and
 get the latest cifsd from here.
 Also, If you have any issue, please send a mail to mailing list or
 register a ISSUE in cifsd in cifsd-team organization.

 Thanks!

**Pull request merged**